### PR TITLE
Refactor ace2 inner

### PR DESCRIPTION
**Pull request by JohnMcLear**
* Remove msie from client side code
* Linting of ace2_inner
* Use strict in most javascript files and bugfix issues discovered in ``use strict``

### CODE REVIEW

**Comment by JohnMcLear** [comment]
@rhansen I'm putting some comments where I'm stuck and need some schooling please :)

**Comment by webzwo0i** [comment]
I pushed a commit to https://github.com/ether/etherpad-lite/commit/06a8a013eeb62103a7b10062ae8bdb1f814216f6 that you can cherry-pick. I just don't wanted to pollute this branch

I would recommend letting handleCut in place if https://github.com/ether/etherpad-lite/issues/2546 and https://github.com/ether/etherpad-lite/pull/2533 are not completely resolved. The comment in the code looks like there was an issue with ordered lists and cut

File: src/static/js/ace2_inner.js [hunk 1]
```diff
@@ -2804,61 +2692,15 @@ function Ace2Inner() {
...
-    // on Windows, always move to start of word.
-    // On Windows, Firefox and IE disagree on whether to stop for punctuation (FF says no).
-    if (browser.msie && forwardNotBack) {
-      while ((!isDone()) && isWordChar(nextChar())) {
-        advance();
-      }
-      while ((!isDone()) && !isWordChar(nextChar())) {
-        advance();
-      }
-    } else {
```
Comment (webzwo0i) [inline src/static/js/ace2_inner.js:2849]: is this dead code? because this branch is executed when it's a non-ie browser

Comment (JohnMcLear) [reply src/static/js/ace2_inner.js:2849]: probably a mistake!

Comment (JohnMcLear) [reply src/static/js/ace2_inner.js:2849]: It's probably a mistake but afaik it was to support a function that was never included or contenteditable supported natively..

File: src/static/js/ace2_inner.js [hunk 2]
```diff
@@ -449,10 +449,8 @@ function Ace2Inner() {
         inCallStackIfNecessary('setStyled', () => {
           fastIncorp(12);
           const clearStyles = [];
-          for (const k in STYLE_ATTRIBS) {
-            if (STYLE_ATTRIBS[k]) {
-              clearStyles.push([k, '']);
-            }
+          for (const k in Object.keys(STYLE_ATTRIBS)) {
```
Comment (JohnMcLear) [inline src/static/js/ace2_inner.js:452]: still errors for me in eslint?  Doesn't for you?

File: src/static/js/ace2_inner.js (hunk 1 above)
Comment (webzwo0i) [reply src/static/js/ace2_inner.js:2849]: I checked this and you are right. It was never used.

File: src/static/js/ace2_inner.js (hunk 2 above)
Comment (webzwo0i) [reply src/static/js/ace2_inner.js:452]: Right. My fault.

**Comment by webzwo0i** [comment]
I've reverted the Object.keys thing and made two more small fixes at 
https://github.com/ether/etherpad-lite/tree/refactor-ace2-inner-fixes

**Comment by JohnMcLear** [comment]
Please tripple check the last commit as I had a lot of conflicts to resolve.

**Pull request merged**